tests/unit/AmalgamatingBusinesses.spec.ts

```diff
@@ -87,15 +93,368 @@ describe.skip('Amalgamating Businesses - add amalgamating business', () => {
+    // verify panel not yet rendered
+    expect(wrapper.find('#add-amalgamating-business-panel').exists()).toBe(false)
+
+    // verify Add button is enabled
+    const addAmalgamatingButton = wrapper.find('#add-amalgamating-business-button')
+    expect(addAmalgamatingButton.classes()).not.toContain('v-btn--disabled')
+
+    // verify Add button and click it
+    expect(addAmalgamatingButton.text()).toBe('Add an Amalgamating Business')
+    await addAmalgamatingButton.trigger('click')
+
+    // verify Add button is now disabled
+    expect(addAmalgamatingButton.classes()).toContain('v-btn--disabled')
+
+    // verify panel is now rendered
+    expect(wrapper.find('#add-amalgamating-business-panel').exists()).toBe(true)
+
+    // verify Cancel button and click it
+    const cancelButton = wrapper.find('#app-cancel-btn')
+    expect(cancelButton.text()).toBe('Cancel')
+    await cancelButton.trigger('click')
+
+    // verify panel is no longer rendered
+    expect(wrapper.find('#add-amalgamating-business-panel').exists()).toBe(false)
+  })
+
+  it('saves an amalgamating business - Extra Pro', async () => {
+    // open panel
+    await wrapper.setData({ isAddingAmalgamatingBusiness: true })
+
+    // simulate saving an extra pro business
+    await wrapper.vm.saveAmalgamatingBusiness({
+      legalType: CorpTypeCd.EXTRA_PRO_A,
+      name: 'Extra Pro Business',
+      identifier: 'A1234567'
+    })
+
+    // verify data
+    expect(store.stateModel.amalgamation.amalgamatingBusinesses.length).toBe(1)
+    const business = store.stateModel.amalgamation.amalgamatingBusinesses[0] as any
+    expect(business.type).toBe(AmlTypes.FOREIGN)
+    expect(business.role).toBe(AmlRoles.AMALGAMATING)
+    expect(business.foreignJurisdiction).toEqual({ country: 'CA', region: 'British Columbia' })
+    expect(business.legalName).toBe('Extra Pro Business')
+    expect(business.corpNumber).toBe('A1234567')
+
+    // verify panel is now closed
+    expect(wrapper.vm.isAddingAmalgamatingBusiness).toBe(false)
+  })
+
+  it('saves an amalgamating business - BC - unaffiliated - non-staff', async () => {
+    // set state
+    store.setKeycloakRoles([])
+    await Vue.nextTick()
+
+    // open panel
+    await wrapper.setData({ isAddingAmalgamatingBusiness: true })
+
+    // mock services functions
+    vi.spyOn((AuthServices as any), 'fetchAuthInfo').mockImplementation(() => null)
+    vi.spyOn((LegalServices as any), 'fetchBusinessInfo').mockImplementation(() => null)
+    vi.spyOn((LegalServices as any), 'fetchAddresses').mockImplementation(() => null)
+    vi.spyOn((LegalServices as any), 'fetchFirstOrOnlyFiling').mockImplementation(() => null)
+
+    // simulate saving a BC business
+    await wrapper.vm.saveAmalgamatingBusiness({
+      legalType: CorpTypeCd.BC_COMPANY,
+      name: 'My BC Business',
+      identifier: 'BC1234567'
+    })
+
+    // verify data
+    expect(store.stateModel.amalgamation.amalgamatingBusinesses.length).toBe(1)
```

Comment:
Can't we use the store getter (`getAmalgamatingBusinesses`) here and below instead Sev?

Reply:
I think so. Want me to change it?

Reply:
No need, was just curious. Resolving this conversation Sev.